packages/studio-ui/src/hooks/useInjectUserStyles.tsx

```diff
@@ -0,0 +1,63 @@
+import { useEffect } from "react";
+import useStudioStore from "../store/useStudioStore";
+
+export default function useInjectUserStyles(
+  iframeDocument: Document | undefined
+) {
+  const [
+    componentTree,
+    getComponentMetadata,
+    activePage,
+    UUIDToFileMetadata,
+    pageCss,
+  ] = useStudioStore((store) => [
+    store.actions.getComponentTree(),
+    store.actions.getComponentMetadata,
+    store.pages.activePageName,
+    store.fileMetadatas.UUIDToFileMetadata,
+    store.pages.getActivePageState()?.cssImports,
+  ]);
+  useEffect(() => {
+    if (!iframeDocument) {
+      return;
+    }
+    componentTree?.forEach((component) => {
+      const cssImports = getComponentMetadata(component)?.cssImports;
+      cssImports?.forEach((cssFilepath) => {
+        injectStyleIntoIframe(iframeDocument, cssFilepath);
+      });
+    });
+    pageCss?.forEach((cssFilepath) => {
+      injectStyleIntoIframe(iframeDocument, cssFilepath);
+    });
+
+    return () => {
+      clearStylingFromIframe(iframeDocument);
+    };
+  }, [
+    componentTree,
+    getComponentMetadata,
+    iframeDocument,
+    pageCss,
+    activePage,
+    UUIDToFileMetadata,
+  ]);
+}
+
+function clearStylingFromIframe(iframeDocument: Document) {
+  const styleElements = [...iframeDocument.head.getElementsByTagName("style")];
+  styleElements.forEach((element) => {
+    element.remove();
+  });
+}
+
+function injectStyleIntoIframe(iframeDocument: Document, filepath: string) {
+  const styletagIdAttribute = `[studio-style-filepath='${filepath}']`;
+  const originalStyletag = document.querySelector(styletagIdAttribute);
```

Comment:
nit: can this variable name be changed to indicate that this is the style tag from the parent document? I was a little confused at first

Reply:
ahh that's a better word for it.  I was struggling to find the right word.  Yes, I'll change that!